backend/main.go

```diff
@@ -79,22 +80,17 @@ func main() {
 		IdleTimeout:  60 * time.Second,
 	}
 
-	go func() {
-		err := srv.ListenAndServe()
-		if err == http.ErrServerClosed {
-			err = nil
-		}
-		if err != nil {
+	terminationCtx := graceful.WithTermination(context.Background())
+
+	defer graceful.Shutdown(terminationCtx, func(err error, exitCode int) {
+		shutdownCtx := context.WithoutCancel(terminationCtx)
```

Comment:
We're almost there!) Let's rethink one little thing: `terminationCtx` is not needed for `srv.Shutdown(terminationCtx)`. So we can simply use `context.Background()` and don't create `shutdownCtx` at all.